### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by paulidale**
All are false positives due to the up_ref/free setup we use.  Added comments to hopefully quiet Coverity.

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
I'll drop 3.0.  I've false positived a *lot* of these "free" problems while dealing with Coverity.  I'm some of these are returns.

**Comment by t-j-h** [comment]
Note: we can also log these with Synopsys as false positives we want to get fixed.

**Comment by paulidale** [comment]
Ping.

**Review by t8m — APPROVED** [review]

File: crypto/evp/exchange.c [hunk 1]
```diff
@@ -420,6 +424,8 @@ int EVP_PKEY_derive_set_peer_ex(EVP_PKEY_CTX *ctx, EVP_PKEY *peer,
                                     EVP_KEYMGMT_get0_name(ctx->keymgmt),
                                     ctx->propquery);
     if (tmp_keymgmt != NULL)
+        /* A Coverity issue with up_ref/down_ref and free */
+        /* coverity[ass_freed_arg] */
```
Comment (slontis) [inline crypto/evp/exchange.c:428]: This one might be missing a p..

**Comment by paulidale** [comment]
Fixed.

**Review by t8m — APPROVED** [review]

File: crypto/evp/exchange.c (hunk 1 above)
Comment (t8m) [reply crypto/evp/exchange.c]: LOL

**Comment by paulidale** [comment]
reping

**Review by slontis — APPROVED** [review]
I also wish to know if  'we want these ugly comments in code?'..
Is this an OTC decision?

**Comment by paulidale** [comment]
We decided to allow some but not a lot.

**Comment by paulidale** [comment]
We decided to allow some but not a lot.

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**